mlir/lib/Transforms/Utils/FoldUtils.cpp

```diff
@@ -318,6 +327,7 @@ OperationFolder::tryGetOrCreateConstant(ConstantMap &uniquedConstants,
     notifyRemoval(constOp);
     rewriter.eraseOp(constOp);
     referencedDialects[existingOp].push_back(dialect);
+    existingOp->setLoc(erasedFoldedLocation);
```

Comment:
Same here.

Reply:
Same logic applies here (see test case `materialize_different_dialect` below).

Though now I think we can make this a bit more explicit by still having the user pass in a location, but if the new location doesn't match the existing location, then we erase it. That way it's more clear what's going on.

Reply:
Actually, in practice it doesn't really make a difference since the only two callers of this private function `tryGetOrCreateConstant` already set the insertion point to the beginning of a block before calling it, so they should already know to ask for an erased location otherwise it's not going to work.